packages/transaction-controller/src/helpers/PendingTransactionTracker.ts

```diff
@@ -320,8 +328,15 @@ export class PendingTransactionTracker {
         return;
       }
 
-      if (receipt?.blockNumber && isSuccess) {
-        await this.#onTransactionConfirmed(txMeta, receipt);
+      const { blockNumber, blockHash } = receipt || {};
```

Comment:
nit: rules out the possibility that `receipt` can be falsy: a little less cognitive load for the reader.

```suggestion
      const { blockNumber, blockHash } = receipt ?? {};
```

Reply:
We typically only use `??` when a falsy value such as `0` is valid, otherwise we default to `||` to handle more invalid cases.

I fear this would achieve the inverse as whenever I see a `??` I assume that some falsy values are expected and allowed.